setup-guide.mdx

```diff
@@ -0,0 +1,167 @@
+
+## Configure Datasources
+
+Having set up a basic project directory, we want to configure the data sources from which we will load data to
+compute our features. We can do so in the Chalk dashboard.
+
+1. In the same directory from before, login or signup with Chalk directly from the command line. The
+[`chalk login`](https://docs.chalk.ai/cli/login) command will open your browser and create an
+API token for your local development, as well as redirect you to your dashboard. If you are not
+redirected you can also find the dashboard at [https://chalk.ai/projects](https://chalk.ai/projects).
+You will automatically see all of the environments in which the email you used to login has been
+provisioned as a user.
+ 2. Within the dashboard, navigate to `Data Sources` in the side bar and add all data sources that you will
+be working with here. After you have saved a data source, you can also use the `Test Data Source` button
+in the upper right hand corner of the Data Source configuration view to verify that your connection is
+valid.
+ 3. Within the working directory, we'll add a `datasources.py` file under our `src` folder to reference the
+data sources that we've added in the dashboard.
+ ```
+ root_directory/
+ ├── src/
+ │  ├── __init__.py
+ │  └── datasources.py
+ ├── chalk.yaml
+ ├── README.md
+ └── requirements.txt
+ ```
+
+Say we added a PostgreSQL data source, then our `datasources.py` might look something like this:
+```python
+pg = PostgreSQLSource(name='PG')
+```
+
+For more details on setting up data sources, [see here.](https://docs.chalk.ai/docs/integrations)
+
+##  Define Feature Sets and Resolvers
+
+Having set up our data sources, we now want to define the registry for this feature store by writing our
+feature sets and resolvers. Each *feature set* is a Python class of features, and a *resolver* tells Chalk
+how to compute the values for different features. Each feature that we write should correspond to a resolver output.
+
+We recommend starting with a minimal feature set and building up iteratively to easily test your code along the way.
+After writing some feature sets, resolvers, and tests, we would expect to see a directory structure like this:
+```
+root_directory/
+├── src/
+│  ├── resolvers/
+│  │  ├── .../
+│  │  ├── __init__.py
+│  │  └── pipelines.py
+│  ├── __init__.py
+│  ├── datasources.py
+│  └── feature_sets.py
+├── tests/
+│  └── ...
+├── .chalkignore
+├── chalk.yaml
+├── README.md
+└── requirements.txt
+```
+
+You can read more in our docs about the [different kinds of features](https://docs.chalk.ai/docs/features) and the
+[different kinds of resolvers](https://docs.chalk.ai/docs/resolver-overview) that you can write. If you would like
+guidance on how to structure your featuresets and resolvers, please reach out in your support channel!
+
+## Deploy and Query
+
+Having defined features and resolvers, the next step is to deploy your pipelines. You can deploy to production by
```

Comment:
same as above, easy to gloss over sentences if they have the same structure as something you've already read

Reply:
facts